plugins/util/html_emitter.py

```diff
@@ -0,0 +1,365 @@
+
+    # module name
+    mod_div += ht.h1(mod.module_name, {"class":"module-name", "id":("mod-" + ht.gen_html_id(mod.module_name))},2,True)
+
+    if mod.module.attrs.has_key('version'):
+      mod_div += ht.h4("openconfig-version: " + mod.module.attrs['version'], {"class":"module-header"},2,True)
+
+    # module description header
+    mod_div += ht.h4("Description", {"class":"module-desc-header"},2,True)
+
+    # module description text
+    paragraphs = text_to_paragraphs(mod.module.attrs['desc'])
+    for para in paragraphs:
+      mod_div += ht.para(para, {"class":"module-desc-text"},2,True)
+
+    mod_div += ht.h4("Imports", {"class":"module-header"},2,True)
+    mod_div += "<p class=\"module-desc-text\">"
+    for i in mod.module.attrs['imports']:
+      mod_div += "%s<br>\n" % i
+    mod_div += "</p>\n"
+
+    mod_div += ht.close_tag(newline=True)
+
+    # initialize and store in the module docs
+    self.moduledocs[mod.module_name] = {}
+    self.moduledocs[mod.module_name]['module'] = mod_div
+    self.moduledocs[mod.module_name]['data'] = ""
+
+    # handle typedefs
+    if len(mod.typedefs) > 0:
+      types_div = ht.open_tag("div", newline=True)
+      types_div += ht.h3("Defined types", {"class":"module-types-header", "id":mod.module_name + "-defined-types"},2,True)
+
+      for (typename, td) in mod.typedefs.iteritems():
+        types_div += ht.h4(typename,{"class":"module-type-name","id":"type-" + ht.gen_html_id(typename)},2,True)
+        types_div += gen_type_info(td.typedoc, 2)
+
+        for prop in YangDocDefs.type_leaf_properties:
+          if td.attrs.has_key(prop):
+            types_div += ht.para(ht.add_tag("span", prop,{"class":"module-type-text-label"}) + ": " + td.attrs[prop],{"class":"module-type-text"},2,True)
+
+        types_div += ht.para(ht.add_tag("span","description:" + ht.br(newline=True), {"class":"module-type-text-label"}) + td.attrs['desc'],{"class":"module-type-text"},2,True)
+
+      types_div += ht.close_tag(newline=True)
+    else:
+      # module doesn't have any typedefs
+      types_div = ""
+
+    # store the typedef docs
+    self.moduledocs[mod.module_name]['typedefs'] = types_div
+
+    # handle identities
+    if len(mod.identities) > 0:
+      idents_div = ht.open_tag("div", newline=True)
+      idents_div += ht.h3("Identities", {"class":"module-types-header", "id":mod.module_name + "-identities"},2,True)
+
+      for base_id in mod.base_identities:
+        idents_div += ht.h4("base: " + base_id,{"class":"module-type-name","id":"ident-" + ht.gen_html_id(base_id)},2,True)
+        idents_div += ht.para(ht.add_tag("span","description:" + ht.br(newline=True), {"class":"module-type-text-label"}) + mod.identities[base_id].attrs['desc'],{"class":"module-type-text"},2,True)
+
+        # collect all of the identities that have base_id as
+        # their base
+        derived = { key:value for key,value in mod.identities.items() if value.attrs['base'] == base_id }
```

Comment:
As mentioned before, I'll try and create a library here that finds all resolved identity names since this won't do multiple inheritance.